### ISSUE TITLE
move to java 8 to use lambda for the routes declaration

### ISSUE THREAD

**Issue by decebals** [decebals/pippo-demo#3]

**Comment by gitblit** [comment]
:+1:

The inline lambdas make short & sweet demos but when your app grows beyond *small* then some different approaches are needed - either lambda handles or full controllers.

I've implemented both approaches and they both have their sweet-spot.

```java
{
    GET("/aRoute", this::aRouteHandler);
    GET("/bRoute", this::bRouteHandler);

    OtherRoutes or = new OtherRoutes();
    GET("/cRoute", or::cRouteHandler);
    GET("/dRoute", or::dRouteHandler);
}

protected void aRouteHandler(RouteContext ctx) {
}

protected void aRouteHandler(RouteContext ctx) {
}
```

**Comment by gitblit** [comment]
Thanks.  We can upstream whatever makes sense.

**Comment by decebals** [comment]
Resolved by #4 PR
